### DISCUSSION TITLE
how to setup .env variable for output path ?

### DISCUSSION THREAD

**Discussion by userchet** [danielmiessler/Fabric#1012] [Q&A]
So I'm on the fabric 2.0, Windows, and the *save* command is not working.
The -o command works but its kinda annoying to always put the full path to the file (obsidian vault).

Here is the question: In the terminal, is there a way so that i can call a variable for the path and only input the name i want for the file ?? 

Thx for helping !!
(btw, I'm a newbie when it comes to coding)

**Comment by Skylightangel** [comment]
If I may ask, what actual command structure did you use for your -o command to manually put it into your vault?
I'm currently using 
`fabric -o=/path/to/obsidian/vault/filename`, which works but has the same issue that you are having.
In the `.env` file located at 
`~/.config/fabric/.env`
I set 
`FABRIC_OUTPUT_PATH="/path/to/obsidian/vault/folder/subfolder/"
`
But it doesn't work sadly :/

**Comment by drego05** [comment]
bumping this -- as I have the same issue/question.

Is the .env naming convention still the same as several months ago? FABRIC_OUTPUT_PATH with the destination in quotes? I'd also like it to save to my Obsidian directory. I suppose if it currently isn't possible - to just have a cron job that files .md files in the working directory and moves them over to my Obsidian vault directory. No rush, just curious. Thanks!

**Comment by verebes1** [comment]
I have just solved this with adding the following into `.zshrc`

```bash
# Define the base directory for Obsidian notes
obsidian_base="/path/to/obsidian"

# Loop through all files in the ~/.config/fabric/patterns directory
for pattern_file in ~/.config/fabric/patterns/*; do
    # Get the base name of the file (i.e., remove the directory path)
    pattern_name=$(basename "$pattern_file")

    # Define a function dynamically for each pattern
    eval "
    $pattern_name() {
        local title=\$1
        local date_stamp=\$(date +'%Y-%m-%d')
        local output_path=\"\$obsidian_base/\${date_stamp}-\${title}.md\"

        # Check if a title was provided
        if [ -n \"\$title\" ]; then
            # If a title is provided, use the output path
            fabric --pattern \"$pattern_name\" -o \"\$output_path\"
        else
            # If no title is provided, use --stream
            fabric --pattern \"$pattern_name\" --stream
        fi
    }
    "
done
```
What this does it creates aliases for all the commands so that you can just use:
`summarize` rather than `fabric --pattern summarize --stream`

If you pass in a title for example `summarize uni_article` then it will create a markdown file with today's date 
`YYYY-MM-DD-uni_article.md` in your obsidian vault path that you provide :) 
Of course you can tweak this to your likings
This has also now been added to the official readme.

**Reply by drego05** [reply]
Score! Excellent. Exactly what I was trying to do
